contribs/gmf/src/directives/layertree.js

```diff
@@ -0,0 +1,301 @@
+goog.provide('gmf.LayertreeController');
+goog.provide('gmf.layertreeDirective');
+
+goog.require('gmf');
+goog.require('ngeo.LayerHelper');
+goog.require('ngeo.LayertreeController');
+goog.require('ngeo.layertreeDirective');
+goog.require('ol.Collection');
+goog.require('ol.layer.Group');
+goog.require('ol.layer.Image');
+goog.require('ol.layer.Tile');
+goog.require('ol.source.ImageWMS');
+
+
+gmfModule.value('gmfLayertreeTemplate',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      var subTemplateUrl = gmf.baseTemplateUrl + '/layertree.html';
+      return '<div ngeo-layertree="gmfLayertreeCtrl.tree" ' +
+          'ngeo-layertree-map="gmfLayertreeCtrl.map" ' +
+          'ngeo-layertree-nodelayer="gmfLayertreeCtrl.getLayer(node)" ' +
+          'ngeo-layertree-listeners="gmfLayertreeCtrl.listeners(treeScope, ' +
+          'treeCtrl)" ' +
+          'ngeo-layertree-templateurl="' + subTemplateUrl + '"' +
+          '</div>';
+    });
+
+
+// Overrides the path to the layertree template (used by each nodes, except
+// the root node that path is defined by the gmfLayertreeTemplate value.
+ngeoModule.value('ngeoLayertreeTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      return gmf.baseTemplateUrl + '/layertree.html';
+    });
+
+
+/**
+ * This directive creates a layertree based on the c2cgeoportal JSON themes
+ * source and a {@link ngeo.layertreeDirective}. The controller used by this
+ * directive defines some fonctions for each node that are created by a default
+ * template. This default template can be overrided by setting the constant
+ * 'gmf.layertreeTemplateUrl' but you will must adapt the
+ * ngeoLayertreeTemplateUrl value too (to define the children's nodes template
+ * path).
+ *
+ * @example
+ * <gmf-layertree
+ *   gmf-layertree-source="ctrl.source"
+ *   gmf-layertree-map="ctrl.map"
+ *   gmf-layertree-wmsurl="ctrl.wmsUrl"
+ * </gmf-layertree>
+ *
+ * @param {string|function(!angular.JQLite=, !angular.Attributes=)}
+ *     gmfLayertreeTemplate Template for the directive.
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfLayertreeDirective
+ */
+gmf.layertreeDirective = function(gmfLayertreeTemplate) {
+  return {
+    scope: {
+      'map': '=gmfLayertreeMap',
+      'getWmsUrlFn': '&gmfLayertreeWmsurl',
```

Comment:
You will remove it to use a constant, but anyway think to use `'='` instead.
Here you have both `'&'` and `::` in your example template, what is redondant.